### ISSUE TITLE
Pass the criteria into beforeUpdate

### ISSUE THREAD

**Issue by cyrilchapon** [balderdashy/sails#5503]
`beforeUpdate` callback would be much more useful if it carried an `originalObject`, as third argument for example.

Retrieving original object is only achievable by either :
- Providing manually, via [policy](http://stackoverflow.com/a/28747257/4578952) or controller, the id in the request body, which is not acceptable
- [Tweaking](http://stackoverflow.com/a/29644240/4578952) and getting raw JS information on the method `this.update.arguments[0]` to get the PK, then `findOne` manually

**EDIT by @dmarcelino:** Given the interesting discussion generated here, we are re-using this thread for a feature request to add the update `criteria` to the `beforeUpdate` lifecycle in a backwards compatible way:

``` javascript
beforeUpdate: function(valuesToUpdate, cb, criteria){}
```

TL;DR: jump to https://github.com/balderdashy/waterline/issues/1004#issuecomment-101699637

See also #1122
